File: StrukturData/Modul2/TugasPraktikum/Kegiatan1.java
package Kuliah.StrukturData.Modul2.TugasPraktikum;

import java.util.ArrayList;

public class Kegiatan1 {
    public static void main(String[] args) {
        ArrayList<String> hewan = new ArrayList<String>();
        hewan.add("Angsa");
        hewan.add("Bebek");
        hewan.add("Cicak");
        hewan.add("Domba");
        hewan.add("Elang");
        hewan.add("Gajah");

        ArrayList<Integer> kosong = new ArrayList<Integer>();

        hewan.add("Badak");
        hewan.add("Bebek");
        int jumlahBebek = 0;
        for (int i = 0; i < hewan.size(); i++) {
            if (hewan.get(i).equals("Bebek")) {
                jumlahBebek++;
                if (jumlahBebek != 0) {
                    kosong.add(i);
                }
            }
        }
        System.out.println("Output No 2:");
        System.out.println(hewan);
        System.out.println("Jumlah Bebek: " + jumlahBebek);
        System.out.println("Posisi Index Bebek: " + kosong + "\n");

        hewan.remove("Bebek");
        System.out.println("Output No 3:");
        System.out.println(hewan + "\n");

        System.out.println("Output NO 4:");
        System.out.println("Elemen pada index ke-0: " + hewan.get(0));
        System.out.println("Elemen pada index ke-2: " + hewan.get(2));
        hewan.remove(0);
        System.out.println(hewan + "\n");

        System.out.println("Output NO 5:");
        hewan.set(0, "Ular");
        System.out.println(hewan);
        hewan.add(2, "Itik");
        System.out.println(hewan + "\n");

        System.out.println("Output NO 6:");
        hewan.subList(1, 5).clear();
        System.out.println(hewan + "\n");
        
        System.out.println("Output NO 7:");
        System.out.println("Elemen pertama: " + hewan.get(0));
        System.out.println("Elemen terakhir: " + hewan.get(hewan.size() - 1) + "\n");

        System.out.println("Output NO 8:");
        System.out.println("Jumlah elemen: " + hewan.size() + "\n");

        System.out.println("Output No 9:");
        int posisiBadak = hewan.indexOf("Badak");
        System.out.println("Posisi index Badak: " + posisiBadak);
    }

}

File: StrukturData/Modul2/TugasPraktikum/Kegiatan1.py
hewan = ["Angsa", "Bebek", "Cicak", "Domba", "Elang", "Gajah"]
kosong = []

hewan.append("Badak")
hewan.append("Bebek")

jumlahBebek = 0
for i in range(len(hewan)):
    if hewan[i] == "Bebek":
        jumlahBebek += 1
        if jumlahBebek != 0:
            kosong.append(str(i))

print("Output No 2:")
print(hewan)
print("Jumlah Bebek:", jumlahBebek)
print("Posisi Index Bebek:", kosong, "\n")

hewan.remove("Bebek")
print("Output No 3:")
print(hewan, "\n")

print("Output No 4:")
print("Elemen pada index ke-0:", hewan[0])
print("Elemen pada index ke-2:", hewan[2])
hewan.pop(0)
print(hewan, "\n")

print("Output No 5:")
hewan[0] = "Ular"
hewan.insert(2, "Itik")
print(hewan, "\n")

print("Output No 6:")
del hewan[1:5]
print(hewan, "\n")

print("Output No 7:")
print("Elemen pertama:", hewan[0])
print("Elemen terakhir:", hewan[-1], "\n")

print("Output No 8:")
print("Jumlah elemen:", len(hewan), "\n")

print("Output No 9:")
posisiBadak = hewan.index("Badak")
print("Posisi index Badak:", posisiBadak)
